### ISSUE TITLE
Unpacking an unknown (or unknown-length) argument into a call leaves no room for additional positional arguments

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3649]
### Summary

I recently ran into this with an `@Todo` from TypeVarTuples, though this applies to any `Unkown`/`Any`/`@Todo` value.
https://play.ty.dev/fe561b85-482f-432d-ae66-c5502abfade5
```py
def foo(x1: int, x2: int, x3: int):
    ...

def bar(x, y: int):
    foo(*x, y)  # Error: Too many positional arguments to function `foo`: expected 3, got 4
```
The expected behavior is that since `x` is `Unkown`/`Any`/`@Todo`, it could expand to any number of any types of input arguments, and ty should choose some combination that makes it succeed

---

This does get a bit more complex though, as in my actual code it was a situation more like this:
https://play.ty.dev/d18da44f-6bd0-4540-90c4-e7f95d9467b6
```py
def foo(x1: int, x2: int, x3: str, x4: int, x5: int):
    ...

def bar(x, y: str, z):
    foo(*x, y, *z)  # Error: Too many positional arguments to function `foo`: expected 5, got 6
```
With a typed value in the middle of two unpacked dynamic values. To make this work, there isn't an easy answer for "how many args should `x`/`z` expand to", since depending on it ty may or may not raise an error on the type of `y`. (Ie in the example, no error if `y` is assigned to `x3`, yes if assigned to any other `x_`). The simplest answer might be that in these ambiguous cases, don't raise any errors for invalid argument types.

---

One other consideration is that there will probably also be people that want situations like my first example, or `foo(y, *x)`, to still raise an error if the type of `y` is "simple" to see that it's wrong (by checking against the type of the first/last function param).

### Version

playground a8a061434

**Comment by dhruvmanila** [comment]
I think we need to apply a similar logic as https://github.com/astral-sh/ruff/pull/22796 for positional call arguments.

**Comment by dhruvmanila** [comment]
And it's not specific to an unknown argument, a `list` will also result in the same outcome:

```py
def foo(x1: int, x2: int, x3: int):
    ...

def bar(x: list[int], y: int):
    foo(*x, y)  # Error: Too many positional arguments to function `foo`: expected 3, got 4
```

[Playground](https://play.ty.dev/3dac6c8a-c0c5-4ca8-bee4-f72f2ea45942)

**Comment by carljm** [comment]
It's interesting that no other type checker does this. I think it should definitely be done for dynamic arguments (as mentioned here), and I think it should be done for length-unknown container types as well (e.g. `list[int]`).

**Comment by MeGaGiGaGon** [comment]
For length unknown ones, I agree with #2628 that there should be a specific diagnostic for it.
I'm surprised no other type checker does this, though I guess it makes sense as with the `@Todo`s my code ends up with way more dynamic types than the normal zero.
